src/app/campaign/store/reducers/account.reducer.ts

```diff
@@ -9,22 +9,16 @@ export interface State extends EntityState<Account> {
 
 export const adapter: EntityAdapter<Account> = createEntityAdapter<Account>();
 
-export const initialState: State = adapter.getInitialState({
-  // additional entity state properties
-});
+export const initialState: State = adapter.getInitialState({});
 
-export function reducer(state = initialState, action: AccountActions): State {
-  switch (action.type) {
-    case ActionTypes.CAMPAIGN_ACCOUNTS_LOAD_SUCCESS: {
-      return { ...adapter.addAll(action.payload.docs.map(docToAccount), state), error: null };
-    }
-    case ActionTypes.CAMPAIGN_ACCOUNTS_LOAD_FAILURE: {
-      return { ...state, error: action.payload.error };
-    }
-    default: {
-      return state;
-    }
-  }
+// tslint:disable-next-line: variable-name
+const _reducer = createReducer(
+  initialState,
+  on(accountActions.AccountsLoadSuccess, (state, action) => ({ ...adapter.addAll(action.docs.map(docToAccount), state), error: null })),
+  on(accountActions.AccountsLoadFailure, (state, action) => ({ ...state, error: action.error }))
+);
+export function reducer(state, action) {
+  return _reducer(state, action);
```

Comment:
This bit is kind of funny looking ... is this a common pattern?  I'm not sure I follow how this is any different than just an `export const reducer = createReducer(...)`

Reply:
It is funny looking, and I don't like it. But this was the problem with the AOT build. https://medium.com/angular-in-depth/solving-aot-error-in-ngrx-function-calls-are-not-supported-in-decorators-5c337381457a

Apparently this goes away with Ivy in ng9